### ISSUE TITLE
[Feature Request] Support MPS Device

### ISSUE THREAD

**Issue by soberhofer** [jacobgil/pytorch-grad-cam#471]
Hello there, 
when trying out grad-cam i noticed that only Cuda is supported for GPU Acceleration.
I have tried adding a `use_mps` parameter to the constructor of the CAM classes, and if set move the model to the `mps` device. 

for me it worked on the following CAM methods : EigenCAM, XGradCAM, GradCAM, HiResCAM, GradCAMPlusPlus, speeding up the runtime in my case by ~10x on my M3 MacBook Pro

ScoreCAM gave me an error, that tensors need to be on the same device. So I think some data would need to be moved to the device as well. 

Has this been looked at before? I could not find anything. 
I could try to draft a PR if there is interest.

**Comment by jacobgil** [comment]
It hasn't,
will be great to have a PR for this.

Do you think it makes sense passing a torch.device object to the constructor, instead of the is_cuda and use_mps ?

**Comment by soberhofer** [comment]
yeah, in my opinion this would be the most "pytorch"-way to do it. 
I'll try to come up with a first PR where i'll switch the boolean `use_cuda` parameter with a `device` parameter of type torch.device

**Comment by jacobgil** [comment]
Merged.
